tests/unit/lib/Any_unittest.cc

```diff
@@ -691,3 +691,65 @@ TEST(Any, lazy_cloneable_visitable)
 	EXPECT_FALSE(any.visitable());
 	EXPECT_THROW(any.visit(nullptr), std::logic_error);
 }
+
+TEST(AnyParameterProperties, old_api_default)
+{
+	AnyParameterProperties params = AnyParameterProperties();
+
+	EXPECT_EQ(
+	    params.get_model_selection(),
+	    EModelSelectionAvailability::MS_NOT_AVAILABLE);
+	EXPECT_EQ(
+	    params.get_gradient(), EGradientAvailability::GRADIENT_NOT_AVAILABLE);
+	EXPECT_FALSE(params.get_model());
+}
+
+TEST(AnyParameterProperties, new_api_default)
+{
+	AnyParameterProperties params = AnyParameterProperties();
+
+	EXPECT_TRUE(params.equals(ParameterProperties()));
+	EXPECT_FALSE(params.has_property(ParameterProperties::MODEL));
+	EXPECT_FALSE(params.has_property(ParameterProperties::GRADIENT));
+	EXPECT_FALSE(params.has_property(ParameterProperties::HYPER));
+}
+
+TEST(AnyParameterProperties, old_custom_ctor)
+{
+	AnyParameterProperties params = AnyParameterProperties(
+	    "test", EModelSelectionAvailability::MS_NOT_AVAILABLE,
+	    EGradientAvailability::GRADIENT_AVAILABLE, true);
+
+	EXPECT_EQ(params.get_description(), "test");
+	EXPECT_EQ(
+	    params.get_model_selection(),
+	    EModelSelectionAvailability::MS_NOT_AVAILABLE);
+	EXPECT_EQ(params.get_gradient(), EGradientAvailability::GRADIENT_AVAILABLE);
+	EXPECT_TRUE(params.get_model());
+
+	EXPECT_TRUE(params.has_property(ParameterProperties::MODEL));
+	EXPECT_TRUE(params.has_property(ParameterProperties::GRADIENT));
+	EXPECT_TRUE(params.equals(
+	    ParameterProperties::MODEL | ParameterProperties::GRADIENT));
+    EXPECT_FALSE(params.equals(ParameterProperties::MODEL));
+	EXPECT_FALSE(params.has_property(ParameterProperties::HYPER));
+}
+
+TEST(AnyParameterProperties, new_custom_ctor)
+{
+	AnyParameterProperties params = AnyParameterProperties(
+	    "test", ParameterProperties::MODEL | ParameterProperties::GRADIENT);
+
+	EXPECT_EQ(params.get_description(), "test");
+	EXPECT_EQ(
+	    params.get_model_selection(),
+	    EModelSelectionAvailability::MS_NOT_AVAILABLE);
+	EXPECT_EQ(params.get_gradient(), EGradientAvailability::GRADIENT_AVAILABLE);
+	EXPECT_TRUE(params.get_model());
+
+	EXPECT_TRUE(params.has_property(ParameterProperties::MODEL));
```

Comment:
I think here we should check that the mask only has *exactly* the two (mode, gradient) set, but nothing else (without saying what the "else" is)
Rather than explicitly testing that bits are not set (again think about adding properties and we would have to extend the test)

Reply:
So just check:
```
EXPECT_TRUE(params.equals(
    ParameterProperties::MODEL | ParameterProperties::GRADIENT));
```

Reply:
Yes, that would do it.